packages/cardano-services/src/Program/programs/providerServer.ts

```diff
@@ -137,17 +145,13 @@ const serviceMapFactory = (options: ServiceMapFactoryOptions) => {
     return new TypeormStakePoolProvider(args, { cache: getDbCache(), connectionConfig$, entities, logger });
   });
 
-  let networkInfoProvider: DbSyncNetworkInfoProvider | undefined;
-
   const getNetworkInfoProvider = (cardanoNode: CardanoNode, dbPools: DbPools) => {
-    if (networkInfoProvider) {
-      return networkInfoProvider;
-    }
+    if (args.useWebSocketApi) return getWebSocketClient().networkInfo;
```

Comment:
Maybe we can get more granularity here.  There might be cases where we have five websocket-enabled providers, but we only want to enable two.

I believe enabling the WebSocket server and using a WebSocket-enabled NetworkInfoProvider can be two different things. Something like:

- `enableWebSocketServer=true`
- `useWebSocketNetworkInfoProvider=true` 
- and provide urls

And, I think using good-old if-else syntax is readable in this kind of program flow. (I know not popular nowadays)

Reply:
Adding configuration variables to be prepared for possible evolutions of the interface doesn't sounds as a good time investment: and if the interface will evolve in a not planned way?
I think it's better to improve the configuration when the interface evolves so the configuration improvement can reflect the interface evolution.

Reply:
I actually thought about making a similar change request, but decided not to as it's a limited scope POC